aten/src/ATen/templates/TensorBody.h

```diff
@@ -195,7 +195,7 @@ class TORCH_API Tensor: public TensorBase {
   //
   // TODO: temporarily disabled
 
-  Tensor& operator=(const TensorBase& x) & {
+  Tensor& operator=(const TensorBase& x) & noexcept {
```

Comment:
These are assignment operations, why is it useful they be marked as noexcept?

Reply:
What about the Scalar assignment ctor down below? Why doesn't that get marked as noexcept too? Bit confused as to why some of these have to be marked as noexcept.

Reply:
I think this is fine, but I would like to hear the motivation why, I want to make sure it doesn't affect any of the overload for the other assignment operators.

Reply:
Marks them noexcept may help some other classes using them as members to become noexcept in their methods.